researchstudio-sat/webofneeds
# BuddySocket and restricting matching by buddy relationship

Adds buddy relationships (back-end and matcher only)
* Adds an ontology for a buddy socket, making `buddy:BuddySocket` available for  `won:socketDefinition` triples.
* Adds a restriction to the embedded queries for the 'Activities' use cases
* Pre-binds the variable `?thisAtom` in embedded queries to the atom that is being matched, such that the matches can be restricted by relationships with the atom. Used for the buddy restrictions above

How to test:
As there is no front-end for creating buddy relationships yet, the relationship has to be added to the matcher's RDF database manually. Then, creating a new Plan should yield the respective matches.
Steps on integration-test:
* create two personas
* create the same interest type for each persona
* create the interest-related plan for one of the personas
* Observe that you don't get any matches
* add the buddy triples manually using blazegraph (change URIs to suit your personas):
```
@prefix buddy: <https://w3id.org/won/ext/buddy#> .
@prefix atom: <https://satvm05.researchstudio.at/won/resource/atom/> .
atom:ub2xfr4vviz7 buddy:buddy atom:wsjfsl5h5xus .
atom:wsjfsl5h5xus buddy:buddy atom:ub2xfr4vviz7 .
```
* create another interest-related plan for one of the personas
* Observe that now you do get matches
